### PR TITLE
[NPU] Support npu op: (1) cos (2) cos_grad

### PR DESCRIPTION
**Pull request by veyron95**
### PR types
New features 

### PR changes
OPs

### Describe
[NPU] Support npu op: (1) cos (2) cos_grad
1. Add test_cos_op_npu.py for Unit tests.
2. Register and implementation of cos npu、cos_grad npu in activation_op_npu.cc.

图片

图片

### CODE REVIEW

File: paddle/fluid/operators/activation_op_npu.cc [hunk 1]
```diff
@@ -431,6 +431,63 @@ class ReciprocalGradNPUKernel : public framework::OpKernel<T> {
   }
 };
 
+template <typename DeviceContext, typename T>
+class CosNPUKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    auto* x = ctx.Input<Tensor>("X");
```
Comment (JiabinYang) [inline paddle/fluid/operators/activation_op_npu.cc:438]: specify const identifier to remind

File: paddle/fluid/operators/activation_op_npu.cc [hunk 2]
```diff
@@ -431,6 +431,63 @@ class ReciprocalGradNPUKernel : public framework::OpKernel<T> {
...
+    auto* out = ctx.Output<Tensor>("Out");
+
+    auto place = ctx.GetPlace();
```
Comment (JiabinYang) [inline paddle/fluid/operators/activation_op_npu.cc:441]: merge this two line together

File: paddle/fluid/operators/activation_op_npu.cc [hunk 3]
```diff
@@ -431,6 +431,63 @@ class ReciprocalGradNPUKernel : public framework::OpKernel<T> {
...
+    out->mutable_data<T>(place);
+
+    auto stream =
+        ctx.template device_context<paddle::platform::NPUDeviceContext>()
+            .stream();
+
+    const auto& runner = NpuOpRunner("Cos", {*x}, {*out}, {});
+    runner.Run(stream);
```
Comment (JiabinYang) [inline paddle/fluid/operators/activation_op_npu.cc:449]: [optional] merge 444 into this

File: paddle/fluid/operators/activation_op_npu.cc [hunk 4]
```diff
@@ -431,6 +431,63 @@ class ReciprocalGradNPUKernel : public framework::OpKernel<T> {
...
+  }
+};
+
+template <typename DeviceContext, typename T>
+class CosGradNPUKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    auto* dout = ctx.Input<Tensor>(framework::GradVarName("Out"));
+    auto* x = ctx.Input<Tensor>("X");
```
Comment (JiabinYang) [inline paddle/fluid/operators/activation_op_npu.cc:458]: same before

File: paddle/fluid/operators/activation_op_npu.cc [hunk 5]
```diff
@@ -431,6 +431,63 @@ class ReciprocalGradNPUKernel : public framework::OpKernel<T> {
...
+            .stream();
+    const auto& runner = NpuOpRunner("Sin", {*x}, {sin_out}, {});
+    runner.Run(stream);
+
+    const auto& runner_dx = NpuOpRunner("Mul", {*dout, sin_out}, {*dx}, {});
+    runner_dx.Run(stream);
+
+    Tensor tmp(x->type());  // Temporary Tensor
+    tmp.Resize(framework::make_ddim({1, 1}));
+    tmp.mutable_data<T>({1}, place);
```
Comment (JiabinYang) [inline paddle/fluid/operators/activation_op_npu.cc]: why `{1}`?

**Review by JiabinYang — COMMENTED** [review]
some comment

File: paddle/fluid/operators/activation_op_npu.cc (hunk 5 above)
Comment (veyron95) [reply paddle/fluid/operators/activation_op_npu.cc]: tmp.mutable_data<T>({1}, place)； means to get a tensor with shape [1]. Maybe it's redundancy, I'll check again.

**Review by veyron95 — COMMENTED** [review]
Redundant {1} has been removed.

File: paddle/fluid/operators/activation_op_npu.cc (hunk 1 above)
Comment (veyron95) [reply paddle/fluid/operators/activation_op_npu.cc:438]: prefer to maintain a style consistent with this .cc file

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**